Testscripts/Linux/ETHTOOL-OFFLOADING-SETTING.sh

```diff
@@ -0,0 +1,268 @@
+            if [[  -z "$server_ip_address"  ]] ; then
+                LogMsg "NIC $SERVER_NIC doesn't have ip even after running dhclient"
+                LogMsg "Server ifconfig $(ip a)"
+                SetTestStateFailed
+                exit 0
+            fi
+        fi
+    done
+    CLIENT_NIC_IPs=$(ssh root@"$VF_IP2" "ip add show | grep -v SLAVE | grep BROADCAST | sed 's/:/ /g' | awk '{print \$2}'")
+    for CLIENT_NIC in "${CLIENT_NIC_IPs[@]}"
+    do
+        client_ip_address=$(ssh root@"$VF_IP2" "ip addr show $CLIENT_NIC | grep 'inet\b'")
+        if [[ -z "$client_ip_address" ]] ; then
+            ssh root@"${VF_IP2}" "pkill dhclient"
+            sleep 1
+            ssh root@"${VF_IP2}" "timeout 10 dhclient $CLIENT_NIC"
+            client_ip_address=$(ssh root@"$VF_IP2" "ip addr show $CLIENT_NIC | grep 'inet\b'")
+            if [[ -z "$client_ip_address" ]] ; then
+                LogMsg "NIC $CLIENT_NIC doesn't have ip even after running dhclient"
+                client_if_config=$(ssh root@"$VF_IP2" "ip a")
+                LogMsg "Client ifconfig ${client_if_config}"
+                SetTestStateFailed
+                exit 0
+            fi
+        fi
+    done
+    LogMsg "Extra NICs have ips"
+fi
+
+# Check if the SR-IOV driver is in use
+VerifyVF
+if [ $? -ne 0 ]; then
+    LogErr "VF is not loaded! Make sure you are using compatible hardware"
+    SetTestStateFailed
+    exit 0
+fi
+
+# Install iPerf3
+ssh -i "$HOME"/.ssh/"$SSH_PRIVATE_KEY" -o StrictHostKeyChecking=no "$remote_user"@"$VF_IP2" ". $HOME/utils.sh && update_repos && install_package iperf3"
+if [ $? -ne 0 ]; then
+    LogErr "Could not install iPerf3 on VM2 (VF_IP: ${VF_IP2})"
+    SetTestStateFailed
+    exit 0
+fi
+
+update_repos
+install_package iperf3
+
+# Start iPerf server on dependency VM
+ssh -i "$HOME"/.ssh/"$SSH_PRIVATE_KEY" -o StrictHostKeyChecking=no "$remote_user"@"$VF_IP2" 'iperf3 -s > perfResults.log &'
+if [ $? -ne 0 ]; then
+    LogErr "Could not start iPerf3 on VM2 (VF_IP: ${VF_IP2})"
+    SetTestStateFailed
+    exit 0
+fi
+
+# Start iPerf client and the time=1800 just make sure the traffic keep running during the whole test
+iperf3 -t 1800 -c "${VF_IP2}" --logfile perfResults.log &
+if [ $? -ne 0 ]; then
+    LogErr "Could not start iPerf3 on VM1 (VF_IP: ${VF_IP1})"
+    SetTestStateFailed
+    exit 0
+fi
+
+sleep 10
+iperf_errors=$(cat perfResults.log | grep -i error)
+if [ $? -eq 0 ]; then
+    LogErr "iPerf3 had errors while running"
+    LogErr "$iperf_errors"
+    SetTestStateFailed
+    exit 0
+fi
+
+# Check if the VF count inside the VM is the same as the expected count
+vf_count=$(find /sys/devices -name net -a -ipath '*vmbus*' | grep pci | wc -l)
+if [ "$vf_count" -ne "$NIC_COUNT" ]; then
+    LogErr "Expected VF count: $NIC_COUNT. Actual VF count: $vf_count"
+    SetTestStateFailed
+    exit 0
+fi
+UpdateSummary "Expected VF count: $NIC_COUNT. Actual VF count: $vf_count"
```

Comment:
Test case1: 2 VF
Test case2: 1VF+1 Synthetic
Test case3: 2 Synthetic

Please post the result or your confirmation that all 3 cases are completed as 'expected'

Reply:
The case is related to SRIOV, so the common scripts/framework are used. Under the common framework, the Test case1 and Test case2 are finished.